hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,264 @@
+                  end points.
+                  '''
+        },
+        {
+            bits: "3",
+            name: "BOOT_REQ_DIS",
+            desc: '''
+                  Setting this bit will disable the feature where the EDN block will automatically
+                  send a boot time request to the CSRNG application interface.
+                  The purpose of this feature is to request entropy as fast as possible after reset,
+                  and during chip boot time.
+                  '''
+        },
+      ]
+    },
+    {
+      name: "SUM_STS",
+      desc: "EDS summary status register",
+      swaccess: "rw",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "0",
+          name: "AUTO_REQ_MODE_SM_STS",
+          desc: '''
+                This bit reflects the status of the AUTO_REQ_MODE state machine. After this mode has
+                been turned off, this status bit should be polled to know when the operation is at
+                a completed state, and it is safe to init registers and set this mode again. A value
+                of zero indicates that the operation is completed.
+                '''
+        }
+        { bits: "31",
+          name: "DIAG",
+          desc: "This this an internal generic diagnostic bit."
+        }
+      ]
+    },
+    {
+      name: "SW_CMD_REQ",
+      desc: "EDN csrng app command request register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwext: "true",
+      hwqe: "true",
+      regwen: "REGEN",
+      fields: [
+        { bits: "31:0",
+          name: "SW_CMD_REQ",
+          desc: '''
+                Writing this reqister with defined CSRNG commands will initiate all
+                possible CSRNG actions. The application interface must wait for the
+                "ack" to return before issuing new commands. This interface is intended
+                to be controlled soley by software. If AUTO_REQ_MODE is set, this
+                register will have no effect on operation.
+                '''
+        }
+      ]
+    },
+    { 
+      name: "SW_CMD_STS",
+      desc: "EDS command status register",
+      swaccess: "rw",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "0",
+          name: "CMD_RDY",
+          desc: '''This bit indicates when the command interface is ready to accept commands.
+                '''
+        }
+        { bits: "1",
+          name: "CMD_ACK",
+          desc: '''This bit is set when the command request has completed. This bit must
+                   be cleared before a new request is made.
+                '''
+        }
+        { bits: "4",
+          name: "CMD_STS",
+          desc: ''' 
+                This one bit field is the status code returned with the application command ack.
+                0b0: Request completed successfully
+                0b1: Request completed with an error
+                '''
+        }
+      ]
+    },
+    {
+      name: "RESEED_CMD",
```

Comment:
i am not quite understanding why the reseed / generate commands are FIFO'd.
Wouldn't the reseed just be sent based on the timer value you outlined below?

I also naively assumed that in auto-request mode, a generate command will be made when the gen-bit FIFO is running low.

Reply:
Is it possible to send parameters with the reseed / generate commands? Reading the document, I assumed you'd have queues like "generate A, generate B" and the auto-request mode would send them in sequence. I don't think there's a CSRNG spec yet(?), so I couldn't check.

Reply:
Realize that there is a lot of flexibility here for CSRNG command generation. FIFOs exist for both generate and reseed commands because they could include additional data - or not. Once a command is sent out, it is re-cycled back into the same respective FIFO for further use (detail). When the glen or generate length is reached, then a reseed must be done, so the EDN block sends out the contents of the reseed FIFO. The exact mechanism for generate requests is that a request is always made, but consumption back-pressure gates fulfillment of the generate if  it is a large request (say 100 words). CSRNG will wait for EDN to consume the last generate word before giving more.

Reply:
Right, this is just allowing EDN's to use nearly the full flexibility of the CSRNG interface.  (Mark is making more work for himself.)